spec/features/merchants/merchants_index_page_spec.rb

```diff
@@ -70,5 +70,47 @@
       click_link(@merchant_1.name)
       expect(current_path).to eq(admin_merchant_path(@merchant_1))
     end
+
+    it 'should enable/disable a merchant after corrisponding button has been clicked' do
+      admin = create(:admin)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+
+      visit merchants_path
+      within "#merchant-#{@merchant_1.id}" do
+        click_link 'Disable'
+        expect(current_path).to eq(merchants_path)
+        #link should now be to enalbe to the account
+        expect(page).to have_link("Enable")
+        @merchant_1.reload
+        expect(@merchant_1.enabled).to eq(false)
+      end
+      #flash message isnt occuring within css tag
+      expect(page).to have_content("#{@merchant_1.name} is now disabled")
+
+      within "#merchant-#{@merchant_2.id}" do
+        click_link 'Disable'
+        expect(current_path).to eq(merchants_path)
+        #link should now be to enalbe to the account
+        expect(page).to have_link("Enable")
+        @merchant_2.reload
+        expect(@merchant_2.enabled).to eq(false)
+      end
+      #flash message isnt occuring within css tag
+      expect(page).to have_content("#{@merchant_2.name} is now disabled")
+    end
   end
+
+  context 'as a merchant' do
+    describe 'attempts to login' do
+      it 'should prevent login if disabled' do
+        merchant = create(:inactive_merchant)
+        visit login_path
+        fill_in "email", with: merchant.email
+        fill_in "password", with: merchant.password
+        click_button "Log In"
+        expect(current_path).to eq(login_path)
```

Comment:
I think it's working but is there any other way to expect that the login failed? Maybe with the flash message you made.

Reply:
User story didn't ask for a flash, but its not a bad idea

Reply:
Well the flash already exists:  `flash[:error] = 'Your credentials are incorrect.'`